docs/developer/architecture/development/index.asciidoc

```diff
@@ -14,16 +11,56 @@ However, these docs will be kept up-to-date to reflect the current implementatio
 ==============================================
 
 [float]
-[[reporting-nav-bar-extensions]]
+=== Reporting Export Types
+"Export Types" are pieces of code that plug into the {kib} Reporting framework, and are responsible for exporting data on behalf
+of a {kib} application. These pieces of code are implemented as TypeScript classes that extend an abstract base class, and
+implement methods for controlling the creation of report jobs, and asynchronously generating report contents. Their `createJob`
+methods handle requests to create report jobs, by accepting jobParams objects and returning "task payload" objects. Their
+`runTask` methods generate the report contents by accepting the task payload object created from the `createJob` function, which
+is then stored in a system index in Elasticsearch.
+
+[float]
+[[reporting-share-service-registrations]]
 === Share menu extensions
-X-Pack uses the `share` plugin of the Kibana platform to register actions in the share menu.
+X-Pack services, such as the {report-features}, register with the `share` plugin of the Kibana platform to register additional
+actions available to make content shareable.
+
+[float]
+=== Generate a report job URL
+To generate a new reporting job, different export types require different `jobParams` objects, which are Rison-encoded and used as
+a `jobParams` query string variable in the Reporting generation endpoint URL. If you use the aforementioned
+<<reporting-share-service-registrations, Sharing plugin registrations>> then this detail will be abstracted away. If your
+application does not use the Share menu extensions, you will have to generate the URL and create a POST request to the URL.
 
 [float]
-=== Generate job URL
-To generate a new reporting job, different export types require different `jobParams` that are Rison encoded into a URL
-that abide by the following convention: `/api/reporting/generate?jobParams=${rison.encode(jobParams)}`. If you use the
```

Comment:
The stated URL convention isn't completely accurate:
* There is now an `internal` version of the path.
* When generating a report in the Kibana UI, the jobParams are given in the POST payload, not the query string